### PR TITLE
Remove support for old versions of Python and Ansible

### LINKED ISSUE #1254: Remove Support for Ansible < 2.15
As part of 3.0.0 cleanup, remove support for Ansible < 2.15 which are all EOL.

### PR DESCRIPTION
**Pull request by pyrodie18**
##### SUMMARY
Resolves #1254

### CODE REVIEW

File: .github/workflows/plugins-integration.yml [hunk 1]
```diff
@@ -25,14 +25,21 @@ jobs:
...
-          - stable-2.13
-          - stable-2.14
           - stable-2.15
           - stable-2.16
           - devel
         python:
+          - '3.9'
           - '3.10'
+          - '3.11'
+          - '3.12'
```
Comment (BGmot) [inline .github/workflows/plugins-integration.yml:35]: This version of Python is not used anywhere in roles tests. Why?

Comment (pyrodie18) [reply .github/workflows/plugins-integration.yml:35]: My thought was that the role aren't using raw python, but the modules are.

Comment (BGmot) [reply .github/workflows/plugins-integration.yml:35]: For roles this is the version of Python that is running Ansible code on Ansible controller.
What's your reasoning for adding Python versions 3.9, 3.11 and 3.12 to modules tests (plugins-integration)? why not just replace 3.10 with the latest 3.12? Do we claim support of particular versions of Python anywhere?

Comment (pyrodie18) [reply .github/workflows/plugins-integration.yml:35]: I think we talk about supporting specific versions of Ansible which in turn ties back to several versions of Python.  I'm not dead set on the change.

**Review by BGmot — APPROVED** [review]

**Pull request merged**